digits/utils/__init__.py

```diff
@@ -152,6 +154,33 @@ def parse_version(*args):
     except AttributeError:
         return pkg_resources.parse_version(v)
 
+def dir_hash(dir_name):
+
+    def file_hash(file_path):
+        md5 = hashlib.md5()
+        blocksize = 64 * 1024
+        with open(file_path, 'rb') as fp:
+            while True:
+                data = fp.read(blocksize)
+                if not data:
+                    break
+                md5.update(data)
+        return md5.hexdigest()
+
+    def reduce_hash(hashlist):
+        md5 = hashlib.md5()
+        for hashvalue in sorted(hashlist):
+            md5.update(hashvalue.encode('utf-8'))
+        return md5.hexdigest()
+
+    if not os.path.isdir(dir_name):
+        raise TypeError('{} is not a directory.'.format(dir_name))
+    hashvalues = []
+    for root, dirs, files in os.walk(dir_name, topdown=True):
+        if not re.search(r'/\.', root):
```

Comment:
Is this supposed to exclude hidden directories? What if the dirname is `.git` instead of `foo/.git`? A comment might be appropriate here, either way.